src/main/webapp/WEB-INF/web.xml

```diff
@@ -1,17 +1,27 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<web-app xmlns="http://xmlns.jcp.org/xml/ns/javaee"
-         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-         xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee http://xmlns.jcp.org/xml/ns/javaee/web-app_4_0.xsd"
-         version="4.0">
 
-    <servlet>
-        <servlet-name>GenericServlet</servlet-name>
-        <servlet-class>Controllers.GenericServlet</servlet-class>
-    </servlet>
-    <servlet-mapping>
-        <servlet-name>GenericServlet</servlet-name>
-        <url-pattern>/</url-pattern>
-    </servlet-mapping>
+<web-app xmlns="http://java.sun.com/xml/ns/javaee"
+	 xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	 xsi:schemaLocation="http://java.sun.com/xml/ns/javaee http://java.sun.com/xml/ns/javaee/web-app_3_0.xsd"
+	 version="3.0">
+    
+    <filter>
+        <filter-name>LoginFilter</filter-name>
+        <filter-class>Filters.LoginFilter</filter-class>
+    </filter>
+    <filter-mapping>
+        <filter-name>LoginFilter</filter-name>
+        <url-pattern>/*</url-pattern>
```

Comment:
Will the addition of the wildcard `*` work in place of the `/index` we introduced in the CSS PR?

Reply:
The /* means the LoginFilter will work on every page, as we need to know where to route somebody when they request any page, not just privileged ones - this is to enable us to redirect /login and /register requests to their dashboard for ergonomics

Reply:
See the above flowchart for the logic of what goes on behind the scenes everytime somebody tries to access any page.